commit a88dfcc1c22750be6ea3283ae08e20019c876234
Author: Dan Carpenter <error27@gmail.com>
Date:   Thu Mar 3 17:56:06 2011 +0100

    keyboard: integer underflow bug
    
    commit b652277b09d3d030cb074cc6a98ba80b34244c03 upstream.
    
    The "ct" variable should be an unsigned int.  Both struct kbdiacrs
    ->kb_cnt and struct kbd_data ->accent_table_size are unsigned ints.
    
    Making it signed causes a problem in KBDIACRUC because the user could
    set the signed bit and cause a buffer overflow.
    
    Signed-off-by: Dan Carpenter <error27@gmail.com>
    Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

diff --git a/drivers/s390/char/keyboard.c b/drivers/s390/char/keyboard.c
index cee4d4e..1160fca 100644
--- a/drivers/s390/char/keyboard.c
+++ b/drivers/s390/char/keyboard.c
@@ -462,7 +462,8 @@ kbd_ioctl(struct kbd_data *kbd, struct file *file,
 	  unsigned int cmd, unsigned long arg)
 {
 	void __user *argp;
-	int ct, perm;
+	unsigned int ct;
+	int perm;
 
 	argp = (void __user *)arg;
 
